### DISCUSSION TITLE
multi-line object

### DISCUSSION THREAD

**Discussion by mmeqbel-wiley** [prettier/prettier#13886] [General]
how can we keep multi-line formatting for destructuring assignment

input: 

```
const {
 a,
 b
} = stuff;
```

output :

`
const { a, b } = stuff;
`

expected: 
```

const {
 a,
 b
} = stuff;
```

**Comment by kachkaev** [comment]
👋 @mmeqbel-wiley! Please share a code snippet as it helps a lot. If I understand your problem correctly, the solution may be to use `//` within `{ }` or `// prettier-ignore` before the statement: [Playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEwA6VOYCGANHgQEZ4C+mAvJhgK4BmzA3HnpFBtpgPT88JTOUw16TVhyhd0WYEWKiBQqBIYwW7TlEGYADgCc4MGAEs4RgLTmA5lAgm5PBWRJlK6upu0yQxCAQBhboyKCERkYQAO4ACpEIaMgghAA2MYQAnsmB5EaEYADWpgDKhAC2cAAy5lBwyMzpaHB5BcVlBoV1dsgwRoytIHAV5HAAJuMT1YRQdoyEdnAAYk4VhGY9KYSMMBABIAAWMBVpAOqH5vBoXWBwpUlX5gBuV1kpYGi5IHUtRjBxAp2daNZpDABWaAAHqUemk4ABFRgQeCgtItQJdIx-FLkQhjNIHYx1GBnczjGCHZAADgADJjoi0zgUDCljHA-s8GoEAI7I+CA4LJFCENDWeoTCYHEx88wmQGLEFIJrooYtCrmPoDNVwxH8hrKsGBGD4skUqlIABMxoK5jSPQAwhAKkrhmgAKwHRgtAAq+OFKoxIGegwAklAprBSmAjOYQgBBCOlGBZeFolpUKhAA)

**Reply by mmeqbel-wiley** [reply]
is it possible to have the expected output without workarounds ?

**Reply by kachkaev** [reply]
The main idea of Prettier is to automate formatting and forget about it completely. Each time we spend mental effort on formatting quirks, we don’t focus on product features and other _more impactful_ things. I know it may feel like I don’t want to answer your question, but what I’m trying to do is to help you ask this question instead: _Do I really want to spend my time on how destructing assignments are formatted?_ 🙂

https://prettier.io/docs/en/why-prettier.html

**Reply by karfous** [reply]
Hello @kachkaev I am not sure if it is planned for future or not, but I would also appreciate to have the possibility for destructuring arrays and object. Don´t you think that adding these // everytime I want prettier to ignore something is the mental effort you are trying to avoid? :) I personally prefer set things up one time in config file and not to use some workarounds during coding.

Code snippet for using require.js
`define(["./cart", "./inventory"], function(cart, inventory) {
       ... some code
);`

In this case I would prefer to have the array expanded, especially if there is 10 and more modules that I deppend on.

**Comment by irkode** [comment]
with prettier 3.5 you added a new `objectWrap` option. and I see the quirks, but the implementation seems to catch a lot of that "object" formatting with either on single line (if short enough) or each value on a line.

I wonder if that isn't true for Arrays as well. Sometimes we have multiple arrays all intended to hold more than one value (or long values)

Imho applying `objectWrap` to Arrays would be something nice:

```JSON
{
  "widgets": {
    "menu": [
      "left",
      "home",
      "right"
    ],
    "sidebar": [
      "about",
      "languages",
      "sidebar_menu",
      "taxonomy_cloud",
      "search"
    ],
    "footer": [
      "justone"
    ]
  }
}
```
which outputs as (even with `object-wrap`"preserve")

```
{
  "widgets": {
    "menu": ["left", "home", "right"],
    "sidebar": [
      "about",
      "languages",
      "sidebar_menu",
      "taxonomy_cloud",
      "search"
    ],
    "footer": ["justone"]
  }
}
```

It's not a problem when writing that stuff, cause it's automatically formatted. but readers will have to switch between inline or multiline depending on content which is harder than following the same style.
